core-web/libs/edit-content/src/lib/store/features/lock.feature.ts

```diff
@@ -0,0 +1,216 @@
+    DotContentletService,
+    DotHttpErrorManagerService,
+    DotMessageService
+} from '@dotcms/data-access';
+import { DotCMSContentlet, DotContentletCanLock } from '@dotcms/dotcms-models';
+
+import { ContentState } from './content.feature';
+import { UserState } from './user.feature';
+
+export interface LockState {
+    lockError: string | null;
+    canLock: boolean;
+    lockSwitchLabel: string;
+}
+
+export const initialLockState: LockState = {
+    lockError: null,
+    canLock: false,
+    lockSwitchLabel: 'edit.content.unlocked'
+};
+
+export function withLock() {
+    return signalStoreFeature(
+        { state: type<ContentState & UserState>() },
+        withState(initialLockState),
+
+        withComputed((store, dotMessageService = inject(DotMessageService)) => ({
+            /**
+             * Determines if the content is currently locked
+             *
+             * @returns boolean - True if content is locked and not in copy locale mode
+             */
+            isContentLocked: computed(() => {
+                const contentlet = store.contentlet();
+                const isCopyingLocale = store.initialContentletState?.() === 'copy';
+
+                return contentlet?.locked && !isCopyingLocale;
+            }),
+
+            /**
+             * Generates a user-friendly message about who has locked the content
+             *
+             * @returns string - Localized message indicating who locked the content or empty string if not locked
+             */
+            lockWarningMessage: computed(() => {
+                const contentlet = store.contentlet();
+                if (!contentlet?.locked) {
+                    return '';
+                }
+
+                const lockedBy = contentlet.lockedBy;
+                const currentUser = store.currentUser();
+                const isLockedByCurrentUser = currentUser.userId === lockedBy.userId;
+
+                const userDisplay = isLockedByCurrentUser
+                    ? dotMessageService.get('edit.content.locked.by.you')
+                    : lockedBy.firstName + ' ' + lockedBy.lastName;
+
+                return dotMessageService.get('edit.content.locked.toolbar.message', userDisplay);
+            })
+        })),
+
+        withMethods(
+            (
+                store,
+                dotContentletService = inject(DotContentletService),
+                dotHttpErrorManagerService = inject(DotHttpErrorManagerService)
+            ) => ({
+                /**
+                 * Locks the current content
+                 *
+                 * Makes an API call to lock the content and updates the store with the returned contentlet
+                 * Clears any previous lock errors before attempting to lock
+                 */
+                lockContent: rxMethod<void>(
+                    pipe(
+                        tap(() =>
+                            patchState(store, {
+                                lockError: null
```

Comment:
we have a loading state?

Reply:
not really a loading state.. this is a visual change in a input switch, and when locked the form will be disabled.